File: modules/TaggedSentenceTest.lua
require 'torch'
require 'nn'

dofile 'Asserts.lua'
dofile 'TaggedSentence.lua'

-- Test nn.TaggedSentence.
local words = {'one', 'two', 'three'}
local tags = {'TAG1', 'TAG2', 'TAG3'}
local sentence = nn.TaggedSentence(words, tags)
assertEquals('one_TAG1 two_TAG2 three_TAG3', tostring(sentence))

-- Test nn.TaggedSentence throws an error for mismatched word and tag table sizes.
words = {'one', 'two'}
tags = {'TAG1'}
local success, message = pcall(nn.TaggedSentence, words, tags)
assert(not success)
assert('TaggedSentence.lua:5: Word count must match tag count!' == message)

File: theano_port/TaggedSentence_test.py
from TaggedSentence import TaggedSentence
import unittest

class TaggedSentenceTest(unittest.TestCase):

  def test_tagged_sentence(self):
    words = ['one', 'two', 'three']
    tags = ['TAG1', 'TAG2', 'TAG3']
    sentence = TaggedSentence(words, tags)
    self.assertEquals('one_TAG1 two_TAG2 three_TAG3', str(sentence))

  def test_mismatched_tag_count(self):
    words = ['one', 'two']
    tags = ['TAG1']
    self.assertRaises(ValueError, TaggedSentence, words, tags)

if __name__ == '__main__':
  unittest.main()
